app/gui2/src/main.ts

```diff
@@ -1,43 +1,77 @@
+import { baseConfig, configValue, mergeConfig } from '@/util/config'
+import { isDevMode } from '@/util/detect'
+import { urlParams } from '@/util/urlParams'
+import { run as runDashboard } from 'enso-authentication'
+import { isOnLinux } from 'enso-common/src/detect'
 import 'enso-dashboard/src/tailwind.css'
 
 const INITIAL_URL_KEY = `Enso-initial-url`
+const SCAM_WARNING_TIMEOUT = 1000
 
-import * as dashboard from 'enso-authentication'
-import { isMac } from 'lib0/environment'
-import { decodeQueryParams } from 'lib0/url'
+let unmount: null | (() => void) = null
+let running = false
 
-const params = decodeQueryParams(location.href)
+function printScamWarning() {
+  if (isDevMode) return
+  const headerCss = `
+    color: white;
+    background: crimson;
+    display: block;
+    border-radius: 8px;
+    font-weight: bold;
+    padding: 10px 20px 10px 20px;
+  `
+    .trim()
+    .replace(/\n\s+/, ' ')
+  const headerCss1 = headerCss + ' font-size: 46px;'
+  const headerCss2 = headerCss + ' font-size: 20px;'
+  const msgCSS = 'font-size: 16px;'
 
-// Temporary hardcode
-const config = {
-  supportsLocalBackend: true,
-  supportsDeepLinks: isMac,
-  shouldUseAuthentication: false,
-  projectManagerUrl: PROJECT_MANAGER_URL,
-  initialProjectName: params.project ?? null,
+  const msg1 =
+    'This is a browser feature intended for developers. If someone told you to ' +
+    'copy-paste something here, it is a scam and will give them access to your ' +
+    'account and data.'
+  const msg2 = 'See https://enso.org/selfxss for more information.'
+  console.log('%cStop!', headerCss1)
+  console.log('%cYou may be the victim of a scam!', headerCss2)
+  console.log('%c' + msg1, msgCSS)
+  console.log('%c' + msg2, msgCSS)
 }
 
-let unmount: null | (() => void) = null
-let runRequested = false
+printScamWarning()
+let scamWarningHandle = 0
+window.addEventListener('resize', () => {
+  window.clearTimeout(scamWarningHandle)
+  scamWarningHandle = window.setTimeout(printScamWarning, SCAM_WARNING_TIMEOUT)
+})
 
 export interface StringConfig {
   [key: string]: StringConfig | string
 }
 
-const vueAppEntry = import('./createApp')
-
 async function runApp(config: StringConfig | null, accessToken: string | null, metadata?: object) {
-  runRequested = true
-  const { mountProjectApp } = await vueAppEntry
-  if (runRequested) {
-    unmount?.()
-    const app = mountProjectApp({ config, accessToken, metadata })
-    unmount = () => app.unmount()
+  running = true
+  const { mountProjectApp } = await import('./createApp')
+  if (!running) return
+  unmount?.()
+  const unrecognizedOptions: string[] = []
+  function onUnrecognizedOption(path: string[]) {
+    unrecognizedOptions.push(path.join('.'))
   }
+  const appConfig = mergeConfig(baseConfig, urlParams(), { onUnrecognizedOption })
```

Comment:
I think here we should merge `config` into `appConfig`. Not sure what is the best ordering though.